File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_VASQUEZ_GABRIEL/Practica 3 Aux/PERSISTENCIA/Ejercicio1/ArchivoEmpleado.java
package Ejercicio1;

import java.io.*;

public class ArchivoEmpleado {
    private String rutaBase;

    public ArchivoEmpleado(String n) {
        this.rutaBase = "C:/Users/Diego/Desktop/mis_empleados/" + n + "/";
        crearArchivo();
    }

    private void crearArchivo() {
        File carpeta = new File(rutaBase);
        if (!carpeta.exists()) carpeta.mkdirs();
    }

    public void guardarEmpleado(Empleado e) throws IOException {
        File archivo = new File(rutaBase + "empleado_" + e.getNombre() + ".txt");
        BufferedWriter bw = new BufferedWriter(new FileWriter(archivo));
        bw.write(e.getNombre() + "\n");
        bw.write(e.getEdad() + "\n");
        bw.write(e.getSalario() + "\n");
        bw.close();
    }

    public Empleado buscaEmpleado(String n) throws IOException {
        File archivo = new File(rutaBase + "empleado_" + n + ".txt");
        if (!archivo.exists()) return null;

        BufferedReader br = new BufferedReader(new FileReader(archivo));
        String nombre = br.readLine();
        int edad = Integer.parseInt(br.readLine());
        float salario = Float.parseFloat(br.readLine());
        br.close();

        Empleado empleado = new Empleado(nombre, edad, salario);

        BufferedWriter bw = new BufferedWriter(new FileWriter(rutaBase + "empleadoBuscado.txt"));
        bw.write(empleado.toString());
        bw.close();

        return empleado;
    }

    public Empleado mayorSalario(float s) throws IOException {
        File carpeta = new File(rutaBase);
        File[] archivos = carpeta.listFiles();
        if (archivos == null) return null;

        Empleado mayor = null;
        for (File archivo : archivos) {
            if (archivo.isFile() && archivo.getName().startsWith("empleado_")) {
                BufferedReader br = new BufferedReader(new FileReader(archivo));
                String nombre = br.readLine();
                int edad = Integer.parseInt(br.readLine());
                float salario = Float.parseFloat(br.readLine());
                br.close();

                if (salario > s) {
                    mayor = new Empleado(nombre, edad, salario);
                }
            }
        }

        if (mayor != null) {
            BufferedWriter bw = new BufferedWriter(new FileWriter(rutaBase + "mayorSalario.txt"));
            bw.write(mayor.toString());
            bw.close();
        }

        return mayor;
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/PERSISTENCIADEOBJETOS/EJERCICIO1/Ej1-PYTHON/ArchivoEmpleado.py
import json
import os
from Empleado import Empleado

class ArchivoEmpleado:
    def __init__(self, nombre_archivo):
        self.__nombre_archivo = nombre_archivo

    def crearArchivo(self):
        """Crea un archivo JSON vacío."""
        with open(self.__nombre_archivo, 'w', encoding='utf-8') as f:
            json.dump([], f, indent=4)

    def guardarEmpleado(self, e):
        lista = self.__leer_lista()
        lista.append({
            'nombre': e.get_nombre(),
            'edad': e.get_edad(),
            'salario': e.get_salario()
        })
        self.__escribir_lista(lista)

    def buscaEmpleado(self, nombre):
        lista = self.__leer_lista()
        for emp in lista:
            if emp['nombre'].lower() == nombre.lower():
                return emp
        return None

    def mayorSalario(self, salario):
        lista = self.__leer_lista()
        for emp in lista:
            if emp['salario'] > salario:
                return emp
        return None

    def __leer_lista(self):
        if os.path.exists(self.__nombre_archivo):
            with open(self.__nombre_archivo, 'r', encoding='utf-8') as f:
                try:
                    return json.load(f)
                except json.JSONDecodeError:
                    return []
        else:
            return []

    def __escribir_lista(self, lista):
        with open(self.__nombre_archivo, 'w', encoding='utf-8') as f:
            json.dump(lista, f, indent=4)
